src/raw_node.rs

```diff
@@ -174,18 +176,50 @@ impl Ready {
     /// If it contains a MsgSnap message, the application MUST report back to raft
     /// when the snapshot has been received or has failed by calling ReportSnapshot.
     #[inline]
-    pub fn messages(&self) -> &Vec<Vec<Message>> {
-        self.light.messages()
+    pub fn messages(&self) -> &[Message] {
+        if !self.is_persisted_msg {
+            self.light.messages()
+        } else {
+            &[]
+        }
     }
 
     /// Take the Messages.
     #[inline]
-    pub fn take_messages(&mut self) -> Vec<Vec<Message>> {
-        self.light.take_messages()
+    pub fn take_messages(&mut self) -> Vec<Message> {
```

Comment:
How about `fn take_messages(&mut self, persisted: bool)`? So that `take_persisted_messages` can be removed.

Reply:
I am ok with these two ways. What's your opinion? @BusyJay

Reply:
Separating them seems more readable. It's strange to call `take_xxx` twice.